docs/assessment.md

```diff
@@ -224,6 +224,109 @@ When calculating a student's grade for a group assessment, PrairieLearn will alw
+| `canAssignRolesAtStart`          | boolean | false   | Allow students with this role to assign roles throughout the assessment. |
+
+Students select their roles before starting an assessment, but they can change their roles mid-assessment if needed. As a safeguard against invalid role configurations, PrairieLearn prevents all question submissions if a group's role configuration does not meet the instructor's specification.
+
+### Adding permissions for an assessment
+
+Permissions can be configured at the _assessment_, _zone_, or _question_ level.
+
+The schema for permissions is defined as follows:
+
+```json
+{
+  "canView": ["Manager", "Reflector", "Recorder", "Contributor"],
+  "canSubmit": ["Recorder"]
+}
+```
+
+| Attribute   | Type            | Default | Description                                                     |
+| ----------- | --------------- | ------- | --------------------------------------------------------------- |
+| `canView`   | Array of string | -       | The names of roles that can view this part of the assessment.   |
+| `canSubmit` | Array of string | -       | The names of roles that can submit this part of the assessment. |
+
+Setting either attribute to `[]` (empty array) means that **no role** can view/submit that part. Setting either attribute to `null` means that **every role** can view/submit that part.
+
+Permissions defined at a higher level are propagated down the assessment hierarchy (assessment -> zone -> question), but permissions defined at lower levels will override those from the higher level. For example, if the permissions in the example above are configured for the assessment, but a question's permissions are configured to `canSubmit: ["Recorder"]`, then only the _Recorder_ can submit that particular question.
+
+### Students joining assessments with custom group roles
+
+When students join a group, they are automatically assigned a role. Students can always view the roles of other students in the group, both before and during an assessment. More information about the assessment's group roles can be expanded in the dropdown menu.
+
+![Joining a group assessment with custom group roles](grouproles_join.png)
+
+When expanded, the dropdown menu shows information about the roles, such as the min/max number of assignments and whether a role can assign other roles.
+
+![Group info from student perspective](grouproles_groupinfo.png)
+
+Any student with an assigner role can view additional controls to change the roles of other users in the group. Students with assigner roles can re-assign group roles both before and during an assessment.
+
+![Group role assignment controls](grouproles_assign_roles.png)
+
+### Permission enforcement during assessment
+
+When an instructor restricts the viewing of a question to certain roles, users without those roles will be unable to view that question.
+
+![Question cannot be viewed due to group role](grouproles_view_question.png)
```

Comment:
can this image be smaller?

Reply:
I'm not sure if you'd like the literal image size to be smaller, or if the screenshot should be less zoomed in. I did the former, but let me know if you meant the latter or something else entirely!

Reply:
I would prefer less zoom (the size of the fonts in the screenshot should be similar to the ones in the text). As it is now, the font size of the image is substantially larger than everything else.

Reply:
@mfsilva22 can you check https://prairielearn--8375.org.readthedocs.build/en/8375/assessment/#assigning-roles-to-students to see if you're ok with the Zoom level used in these new screenshots?